### ISSUE TITLE
.NETCORE 2.0 AcquireTokenAsync throwing System.AggregateException. TaskCanceledException: A task was canceled

### ISSUE THREAD

**Issue by rishavraj10** [AzureAD/azure-activedirectory-library-for-dotnet#1239]
**Which Version of ADAL are you using ?**
ADAL 3.16.4

**Which platform has the issue?**
netcore

**What authentication flow has the issue?**
* Desktop 
    * [ ] Interactive
    * [ ] WIA
    * [ ] U/P
    * [ ] Device code flow (browserless)
* [ ] Mobile
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [ ] OBO

Other? - Linux Daemon

**Repro**

```csharp
var authenticationContext = new AuthenticationContext(auth);
if (authenticationContext == null)
throw (new ArgumentNullException("AuthenticationContext")); authenticationContext.TokenCache.Clear();
ClientAssertionCertificate clientAssertion = new ClientAssertionCertificate(clientId, certificate);

AuthenticationResult authenticationResult = authenticationContext.AcquireTokenAsync(audience, clientAssertion).Result;
```

**Expected behavior**
The code is expected to return the token. We are using cert based authentication for fetching the token.

**Actual behavior**
The Fetch token call is failing after a time out of 1 minute with TaskCancelledException. Below is the exception:

System.AggregateException: One or more errors occurred. (unknown_error: Unknown error) ---> Microsoft.IdentityModel.Clients.ActiveDirectory.AdalServiceException: unknown_error: Unknown error ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.ConfiguredTaskAwaitable1.ConfiguredTaskAwaiter.GetResult() at System.Net.Http.HttpClient.<FinishSendAsyncBuffered>d__58.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.IdentityModel.Clients.ActiveDirectory.HttpClientWrapper.<GetResponseAsync>d__30.MoveNext() --- End of inner exception stack trace --- at Microsoft.IdentityModel.Clients.ActiveDirectory.AdalHttpClient.<GetResponseAsync>d__211.MoveNext()

Can you suggest some possible cause for the issue?

**Possible Solution**

**Additional context/ Logs / Screenshots**
Add any other context about the problem here, such as logs and screebshots. Logging is described at https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/wiki/Logging-in-ADAL.Net

**Comment by jmprieur** [comment]
@rishavraj10 : 
- did you try upgrading to ADAL 3.19.8 ? We fixed the following issue in 3.19.2 : [ADAL.NET 3.19.2 fixes the "user canceled the authentication" intermittent error on the UWP platform](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/adalnet_3_19#adalnet-3192-fixes-the-user-canceled-the-authentication-intermittent-error-on-the-uwp-platform)
- would you have some threading issues (I see that you are using `.Result` which is not recommended (See [Reminder: Do not call ADAL or MSAL async methods synchronously from the UI thread](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/adalnet_3_19#reminder-do-not-call-adal-or-msal-async-methods-synchronously-from-the-ui-thread) for more details)
- is it something that has ever worked? or is it recent?
- would you have network issues?
- Would you be using a proxy?  if that's the case, did you have a look at [Using ADAL with a proxy](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/wiki/with-proxy)

**Comment by bgavrilMS** [comment]
Closing, please upgrade to ADAL 3.19.8 or ADAL 4.0.0-preview and try the other advice provided.
